test/plugin/base_test.rb

```diff
@@ -2292,9 +2420,22 @@ def verify_subfields_when_not_hash(payload_key)
       d.run
     end
     verify_log_entries(1, COMPUTE_PARAMS, 'jsonPayload') do |entry|
-      field = get_fields(entry['jsonPayload'])[payload_key]
-      assert_equal 'a_string', get_string(field), entry
-      assert_false entry.key?(destination_key), entry
+      if payload_key == DEFAULT_LABELS_KEY
```

Comment:
This seems like two separate methods in one: `verify_subfields_removed_when_not_hash` and `verify_subfields_untouched_when_not_hash`.

Reply:
Done